### DISCUSSION TITLE
Bracket Order With Market Price

### DISCUSSION THREAD

**Discussion by michaelkawaicheung** [erdewit/ib_insync#475] [General]
Does anyone knows how to enter a market order with bracket order (take profit and stop loss) attached? It can be done in TWS but seems that not in ib_insync. I have tried `bracketOrder` but it requires a limit order price.
![image](https://user-images.githubusercontent.com/66937044/167267063-753d4d65-a13e-434f-b6ed-b2ba452a5e15.png)

**Comment by ronniereid** [comment]
I would love this option too!

**Comment by August1328** [comment]
That is actually quite simple. You have got your standard bracket order code and just add an extra line which defines the parent order type before execution:

`bracket = ib.bracketOrder(...)`
`bracket.parent.orderType = "MKT`
`for order in bracket:`
`    ib.placeOrder(contract, order)`

Works for me ordering options and should also work for stocks.
